src/Control/Varying/Core.hs

```diff
@@ -83,15 +99,15 @@ import Debug.Trace
 --------------------------------------------------------------------------------
 -- | Lift a pure computation into a stream.
 var :: Applicative m => (a -> b) -> VarT m a b
-var f = VarT $ \a -> pure (f a, var f)
+var f = VarT $ \(!a) -> pure (f a, var f)
```

Comment:
It made a big difference in runSplineT so I propagated that change to all similar patterns, but then I removed runSplineT! I'll have to remove these and do a profiling run to compare. I'd like to not have all the bangs hanging around if i can help it.